src/main/antora/modules/security/pages/security.adoc

```diff
@@ -1713,4 +1713,222 @@ The test itself is basically the same as that for the <<Securing an endpoint wit
+    loginToContinue = @LoginToContinue(
+        loginPage="/login.xhtml",
+        errorPage=""
+    )
+)
+@DeclareRoles({ "user", "caller" })
+@FacesConfig
+@ApplicationPath("/rest")
+public class ApplicationConfig extends Application {
+
+}
+----
+
+include::partial$declare-authentication-mechanism.adoc[]
+
+Contrary to the Basic HTTP authentication mechanism, the Form authentication mechanism allows us to customize the login dialog and to keep track of the authenticated session on the server (using a cookie). This also allows us to logout, something that for unknown reasons has never been specified for Basic HTTP authentication.
+
+Contrary to the regular Form authentication mechanism, the Custom Form authentication mechanism let's us customize the login dialog even more by having the ability to execute custom code between the postback of a login form and the form authentication mechanism taking the provided credentials.
+
+To use this authentication method, we need to designate a path to a resource that is relative to our application. The authentication mechanism will redirect the caller to this resource when authentication is required. The resource can be anything, but a postback should eventually lead to some code being executed that continues the authentication dialog. E.g. a plain .html file or .jsp file combined with  a Filter, or a Faces view with a backing bean.
+
+
+==== Define the authentication mechanism's view and backing code
+
+For this example we'll use a Faces view with a backing bean:
+
+[source,xml]
+----
+<html lang="en"
+    xmlns="http://www.w3.org/1999/xhtml"
+    xmlns:h="jakarta.faces.html"
+    xmlns:faces="jakarta.faces"
+>
+    <h:head/>
+    <h:messages />
+    
+    <body>
+        <p>
+            Login to continue
+        </p>
+    
+         <form faces:id="form">
+            <p>
+                <strong>Username </strong> 
+                <input faces:id="username" type="text" faces:value="#{loginBacking.username}" />
+            </p>
+            <p>
+                <strong>Password </strong> 
+                <input faces:id="password" type="password" faces:value="#{loginBacking.password}" />
+            </p>
+            <p>
+                <input type="submit" value="Login" faces:action="#{loginBacking.login}" />
+            </p>
+        </form>
+    </body>
+</html>
+----
+
+[source,java]
+----
+@Named
+@RequestScoped
+public class LoginBacking {
+
+    @Inject
+    private SecurityContext securityContext;
+
+    @Inject
+    private FacesContext facesContext;
+
+    @NotNull
+    @Size(min = 3, max = 15, message="Username must be between 3 and 15 characters")
+    private String username;
+
+    @NotNull
+    @Size(min = 5, max = 50, message="Password must be between 5 and 50 characters")
+    private String password;
+
+    public void login() {
+        switch (
+            // Continue the authentication dialog manually by invoking the authenticate()
```

Comment:
When I read "dialog" I imagine a dialog box..... maybe say "process"?

Reply:
A bit of the problem with that is that "authentication dialog" has already been established in the specification (including the javadoc). It's the dialog between the caller and the security system.

Reply:
OK. Have you established that previously in the tutorial?

Reply:
I'll go through it, and see if it needs some explanation. Possibly in a note.